src/GraphQL.Tests/Bugs/Bug1773.cs

```diff
@@ -51,40 +51,45 @@ public void nonnull_list_valid()
         [Fact]
         public void nonnull_list_throws_when_null()
         {
-            AssertQueryWithError("{testListNullInvalid}", "{\"testListNullInvalid\": null}", "Error trying to resolve field 'testListNullInvalid'.", 1, 2, new[] { "testListNullInvalid" },
-                new InvalidOperationException("Cannot return a null member within a non-null list for list index 0."));
+            AssertQueryWithError("{testListNullInvalid}", "{\"testListNullInvalid\": null}", "Error trying to resolve field 'testListNullInvalid'.", 1, 2, new object[] { "testListNullInvalid", 0 },
+                new InvalidOperationException("Cannot return null for a non-null type. Field: testListNullInvalid, Type: Int!."));
         }
 
         [Fact]
         public void list_throws_for_invalid_type()
         {
-            // TODO: does not yet fully meet spec (does not return members of lists that are able to be serialized, with nulls and individual errors for unserializable values)
-            AssertQueryWithError("{testListInvalidType}", "{\"testListInvalidType\": null}", "Error trying to resolve field 'testListInvalidType'.", 1, 2, new[] { "testListInvalidType" },
-                new FormatException("Input string was not in a correct format."), localizedMessage: "Входная строка имела неверный формат.");
+            AssertQueryWithError("{testListInvalidType}", "{\"testListInvalidType\": [ null ]}", "Error trying to resolve field 'testListInvalidType'.", 1, 2, new object[] { "testListInvalidType", 0 },
```

Comment:
The same "problem" here as https://github.com/graphql-dotnet/graphql-dotnet/pull/2341/files#r588111945 .
`testListInvalidType` is nullable array of nullable enums. `Error trying to resolve field 'testListInvalidType'` seems confusing a little since `testListInvalidType` field was successfully returned. I understand that in the case of arrays, we do not have an array element name. There is only an index. We now see this index in `Path`. It's good. But it is bad that an error message can be perceived completely differently.

I would prefer to see an error message like
`Error trying to resolve array element for field 'testListInvalidType'.` (**array element for** added)

Reply:
We can work on this, but it should be done within a separate PR.  This PR is getting too big, and it will be hard to understand the consequences of additional changes.

Reply:
Agree.